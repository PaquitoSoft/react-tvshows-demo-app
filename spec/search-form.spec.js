import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchForm from '../src/app/search-form';

const noop = () => false;

describe('SearchForm', () => {

	test('Should render the form', () => {
		const component = shallow(<SearchForm onSubmit={noop} />);
		expect(component.find('form').length).toBe(1);
		expect(component.contains(<button type="submit">List repos</button>)).toBe(true);
	});

	test('Should render the form (snapshot)', () => {
		const component = shallow(<SearchForm onSubmit={noop} />);
		const componentJson = shallowToJson(component);
		expect(componentJson).toMatchSnapshot();
	});

});
