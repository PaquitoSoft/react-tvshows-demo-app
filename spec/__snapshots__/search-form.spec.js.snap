// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Should render the form (snapshot) 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    name="username"
    placeholder="Type a GitHub user account name"
    size="40"
    type="text"
  />
  <button
    type="submit"
  >
    List repos
  </button>
</form>
`;
